src/chatlog/chatmessage.cpp

```diff
@@ -184,8 +184,11 @@ QString ChatMessage::detectAnchors(const QString &str)
                 "(?:\\b)(www\\.|http[s]?|ftp)://" // (protocol)://(printable - non-special character)
                 // http://ONEORMOREALHPA-DIGIT
                 "\\w+\\S+)" // any other character, lets domains and other
-                "|(^tox:[@\\w]+$)"); // starts with `tox` then : and only alpha-digits till the end
-                // also accepts tox:agilob@net as simplified TOX ID
+                "|"   // standard protocols OR tox ID:
+                "^tox:(([\\w]+@[\\S]+)" // tox:something@something.com
+                "|(\\w{76}))$" // tox:76 characters long tox id
```

Comment:
@agilob Uhm ... forget what I wrote above. Did you know about `QRegularExpression`? Looks like this replaced the `QRegExp` class in Qt5 and provides what you are looking for.

Apart from that I agree with @zetok in this particular case. In general we do not want to use "hacking tricks" to improve speed. So `[0-9a-fA-F]{76}` would be ok.

Btw.: If reg. exp. are to slow, we can use the classic way and parse by ourselves:

```c++
// pretty shitty sample parser - just to make clear what I mean :)

QString toxId;
QString destination;
bool ok;
if (string.startsWith("tox:"))
    int pos = string.indexOf('@');
    if (pos > -1) {
        toxId = string.mid(4, pos).toInt(&ok, 16);
        if (ok) {
            destination = string.right();
        }
    }
}
```